### DISCUSSION TITLE
Twitch and Prime crash

### DISCUSSION THREAD

**Discussion by wikked420** [aclap-dev/video-downloadhelper#354] [Help and Q&A]
### Your question

Hi,

I've been having a problem with the app for a while, before I switched to windows 10 last year ( I know ) I managed to fix the issue with inserting Twitch into the black list of the app, but moving to windows 10 doesn't fix the issue anymore of what happens to me.

I usually put Twitch on background; not usually saving vods there, but I have it on and I occasionally watch Prime videos, now the issue I get is when and if I close out the tab, or even refresh the browser the whole browser will crash on me until I forcefully close it out. I have tried many ways to add the sites to blacklist like before, but it doesn't work anymore; aside from closing the app when not in use can someone shed some light on this issue? 

                                                                             Thank you!

### Addon & CoApp version and details

version: 8.2.2.8
channel: stable
build date: 2024-03-18
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:124.0) Gecko/20100101 Firefox/124.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\Shiela",
  "converterError": "Exit without answer"
}

**Comment by Mattmjs** [comment]
Maybe you should not have Twitch open in the background until you actually want to view something. 
You can unload twitch as a background tab if it is the least used tab from _about:unloads_
It needs to be in the priority 1 position. What would be even better is if Mozilla would allow extensions to run per tab instead of all tabs.
Chrome and Edge do this. You could try suggesting this to Mozilla.

**Reply by Steffronte** [reply]
Got the same issue and figured out it was due to this addon thanks to an 4 years old reddit post about this issue : https://www.reddit.com/r/firefox/comments/hz25ek/firefox_freezes_and_crashes_when_closing_twitch/

I fixed it by disabling the addon when I don't need it, since like the author, I also need to have twitch in the background.

**Reply by GrampaWildWilly** [reply]
Nobody has pointed out that 8.2.2.8 & 2.0.10 are woefully out of date.  Oops.  I guess I have now.

**Comment by tunacanman** [comment]
For me it crashes on it's own, not just when refreshing or closing the tab. I've had this issue since switching to Firefox from Chrome(Chrome VHD didn't have this problem). I always have a Twitch stream in the background. I'm sure that's a common practice. After a handful of hours or so, the browser freezes then crashes. I disabled VDH and I haven't crashed since then, a few weeks ago.
Did the blacklist feature I hear about around here get removed? 
I can't find an extension that toggles extensions based on website--I guess bc Firefox doesn't have per-tab functionality for that.

Edit: More behavior info: I enabled the extension again yesterday. This morning(PC and browser left on over night) when switching to a Twitch tab, the browser froze, but after waiting a minute it managed to switch to a new tab and I was able to close the problematic tab. 
Edit: Had a freeze when trying to close a Twitch tab. Again, waiting it out avoided a crash, duplicated and closed the bad tab.

version: 9.1.0.44
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:132.0) Gecko/20100101 Firefox/132.0
{
    "download_directory": "D:\\Downloads WD",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 4,
    "successfull_dl": 38
}
